src/ScaleformMCM.cpp

```diff
@@ -207,18 +218,40 @@ namespace ScaleformMCM {
 
 				VirtualMachine* vm = (*G::gameVM)->m_virtualMachine;
 
-				VMValue senderValue;
-				PackValue(&senderValue, &targetForm, vm);
-
-				if (!senderValue.IsIdentifier()) {
+				IObjectHandlePolicy * hp = (*g_gameVM)->m_virtualMachine->GetHandlePolicy();
+				UInt64 hdl = hp->Create(targetForm->kTypeID, targetForm);
+
+				VirtualMachine::IdentifierItem * attachedScripts = (*g_gameVM)->m_virtualMachine->m_attachedScripts.Find(&hdl);
+
+				if (attachedScripts->count == 0)
+				{
+					_WARNING("Warning: Cannot run function %s on a form with no scripts attached.", args->args[2].GetString());
+					return;
+				}
+
+				VMIdentifier* targetScriptIdentifier = nullptr;
+
+				if (attachedScripts->count == 1)
+				{
+					targetScriptIdentifier = attachedScripts->GetScriptObject(attachedScripts->identifier.one);
+				}
+				else
+				{
+					for (UInt32 i = 0; i < attachedScripts->count; ++i)
+					{
+						targetScriptIdentifier = attachedScripts->GetScriptObject(attachedScripts->identifier.many[i]);
+						if (_stricmp(targetScriptIdentifier->m_typeInfo->m_typeName.c_str(), args->args[1].GetString()) == 0)

```

Comment:
This large block for getting VM identifiers by script name is duplicated three times, can it be simplified and de-duplicated?